### PR TITLE
model objects for Submission and Workflow with CRUD

### PR DESCRIPTION
**Pull request by tedsharpe**
Submission, Workflow, and CRUD API defined

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 1]
```diff
@@ -68,7 +68,7 @@ trait GraphDAO {
   // named PipeFunctions
 
   def hasProperty[T](key: String, value: T) = new PipeFunction[Vertex, java.lang.Boolean] {
-    override def compute(v: Vertex) = v.getProperty[T](key).equals(value)
+    override def compute(v: Vertex) = v.getProperty[T](key) == value
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: what's the difference here?

Comment (tedsharpe) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: NullPointerException if the property doesn't exist.

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/JobApiService.scala [hunk 2]
```diff
@@ -35,10 +35,10 @@ trait JobApiService extends HttpService with PerRequestCreator {
   def submitJobRoute = cookie("iPlanetDirectoryPro") { securityTokenCookie =>
     path("workspaces" / Segment / Segment / "jobs") { (workspaceNamespace, workspaceName) =>
       post {
-        entity(as[JobDescription]) { jobDesc =>
+        entity(as[Submission]) { submission =>
           requestContext => perRequest(requestContext,
                                        WorkspaceService.props(workspaceServiceConstructor),
-                                       WorkspaceService.SubmitJob(WorkspaceName(workspaceNamespace,workspaceName),jobDesc,securityTokenCookie))
+                                       WorkspaceService.SubmitJob(WorkspaceName(workspaceNamespace,workspaceName),submission,securityTokenCookie))
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/JobApiService.scala]: rename WorkspaceService.SubmitJob to WorkspaceService.CreateSubmission?

File: src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * @author tsharpe
+ */
+
+// this is what a client sends to have us create a submission
+
+// Request for a submission
+@ApiModel(value = "Submission")
+case class Submission(
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala]: can we rename this SubmissionRequest?

File: src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+// Cromwell's response to job submission (not decorated with annotations because it's not part of our API)
+case class JobStatus(
+  id: String,
+  status: String
+)
+
+// Status of a workflow
+@ApiModel(value = "WorkflowStatus")
+case class WorkflowStatus(
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala]: can we rename this Workflow? I want to make a WorkflowStatus enum that contains the actual status values and having this be called WorkflowStatus makes it awkward.

File: src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  status: String,
+  @(ApiModelProperty@field)("Status last-changed date") @(VertexProperty@field)
+  statusLastChangedDate: DateTime,
+  @(ApiModelProperty@field)("Entity name") @(VertexProperty@field)
+  entityName: String
+)
+
+// Status of a submission
+@ApiModel(value = "SubmissionStatus")
+case class SubmissionStatus(
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala]: can we rename this Submission?

**Comment by tedsharpe** [comment]
renaming has occurred

**Pull request merged**